test/integration/dns_test.go

```diff
@@ -62,6 +62,37 @@ func TestDNS(t *testing.T) {
 		close(stop)
 	}, 50*time.Millisecond, stop)
 
+	// verify service DNS entry is visible via master service
```

Comment:
@smarterclayton can you verify this is reasonable? I haven't dug into the dns bits enough to know